File: credit-engine-api/src/main/java/com/credit/engine/application/service/currency/ExchangeRateService.java
package com.credit.engine.application.service.currency;

import com.credit.engine.application.dto.currency.ExchangeRateRequest;
import com.credit.engine.application.dto.currency.ExchangeRateResponse;

public interface ExchangeRateService {

    ExchangeRateResponse create(ExchangeRateRequest request);

    ExchangeRateResponse findLatestRate(String originCurrencyCode, String destinationCurrencyCode);

}

File: credit-engine-ui/src/services/currency/exchangeRate.service.ts
import { http } from '../../api/client';
import type ExchangeRateRequest from './dto/ExchangeRateRequest';
import type ExchangeRateResponse from './dto/ExchangeRateResponse';

const controllerUrl = '/exchange-rates';

export const exchangeRateService = {

    create(request: ExchangeRateRequest) {
        return http.post<ExchangeRateResponse>(controllerUrl, request);
    },

    findLatestRate(origin: string, destination: string) {
        return http.get<ExchangeRateResponse>(
            `${controllerUrl}/latest?origin=${origin}&destination=${destination}`
        );
    },

};
